### PR TITLE
WIP: Protobuf v2 with C++ support

### PR DESCRIPTION
**Pull request by jakirkham**
This proposes to build Python and C++ support for protobuf. Thus it can include things like the protobuf compiler. Here we use the most recent 2.x version.

* [x] Build the protobuf compiler `protoc`.
* [x] Build C++ support.
* [ ] Add back Windows support.
* [ ] Explore Python-C++ bindings.
* [x] Merge PR ( https://github.com/conda-forge/staged-recipes/pull/171 )
* [x] Merge PR ( https://github.com/conda-forge/staged-recipes/pull/172 )

### CODE REVIEW

**Comment by pelson** [comment]
:+1: - once you have the green ticks, I'm happy for v2 to be merged into this branch.

**Comment by jakirkham** [comment]
Alright, I am going to restart the CI now that we have the dependencies here.

**Comment by jakirkham** [comment]
Very strange failure on Circle CI related to install/build of gtest. Maybe it makes sense to just package gtest and make it a dependency to avoid this? We may want it for other reasons anyways.

**Comment by jakirkham** [comment]
Seems installation of Google Tests is [verbotten]( https://github.com/google/googletest/blob/release-1.7.0/Makefile.am#L296-L298 ).

**Comment by jakirkham** [comment]
Looks like we are not alone CockroachDB saw [this problem]( https://groups.google.com/forum/#!msg/cockroach-db/MrrWh4cAZcY/Zg-yXEe_on4J ) before.

**Comment by jakirkham** [comment]
Appears they addressed this problem in [here]( https://github.com/cockroachdb/cockroach/pull/162 ) somehow.

**Comment by jakirkham** [comment]
Appears that CockroachDB addressed this problem by simply [not running]( https://github.com/cockroachdb/cockroach/pull/162/files#diff-00c1868329bff6ba766f2caefa0451dfR50 ) `make check` on CircleCI. Unfortunately, I think we will have to do the same thing here.

File: recipe/build.sh [hunk 1]
```diff
@@ -1,6 +1,18 @@
...
+./configure --prefix=$PREFIX
+make
+
+# Cannot run `make check` on CircleCI.
+# See the PR that added this for details.
+# https://github.com/conda-forge/protobuf-feedstock/pull/5
+if [[ ! ( $CIRCLECI ) ]];
+then
+    make check
+fi
```
Comment (jakirkham) [inline recipe/build.sh]: We are unable to run this on CircleCI. See [log]( https://circleci.com/gh/conda-forge/protobuf-feedstock/18 ) for `make check` failure. This problem is unique to Circle CI and protobuf, but [others have suffered]( https://github.com/cockroachdb/cockroach/pull/162#issuecomment-62233187 ) from it too. The only [fix]( https://github.com/cockroachdb/cockroach/pull/162/files#diff-00c1868329bff6ba766f2caefa0451dfR50 ) appears to be not to run `make check`. So, that is what we do here.

**Comment by jakirkham** [comment]
Not running `make check` doesn't seem to matter as it is still downloading gtest.

**Comment by jakirkham** [comment]
Switched to a source that was packaged up. Thus it includes `gtest` so we no longer need to download it. Had some configuration errors where it expects `automake` 1.14.1, but our version is 1.14. So, had to to go through `aclocal`, `automake`, and `autoconf`. This had to be done for the source directory and `gtest` within it. This all works so that we can run `make check` again.

**Comment by jakirkham** [comment]
However, we are seeing one test failure on Linux, which we will need to investigate. Message below.

    FAIL: google/protobuf/compiler/zip_output_unittest.sh

**Comment by jakirkham** [comment]
Closed out as the v3 releases have been pretty reliable and a non-beta appears to be on its way out.

**Pull request closed without merging**